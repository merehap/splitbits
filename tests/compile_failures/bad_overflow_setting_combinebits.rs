use splitbits::*;

fn main() {
    let a: u8 = 0b1010_0101;
    combinebits!(overflow=explode, "aaaa aaaa");
}
