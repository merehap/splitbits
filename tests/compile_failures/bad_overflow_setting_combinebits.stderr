error: proc macro panicked
 --> tests/compile_failures/bad_overflow_setting_combinebits.rs:5:5
  |
5 |     combinebits!(overflow=explode, "aaaa aaaa");
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: called `Result::unwrap()` on an `Err` value: "'explode' is an invalid overflow option. Options: 'truncate', 'panic', 'corrupt', 'saturate'."
